### PR TITLE
chore(markdownlint) add emphasis style rule

### PR DESCRIPTION
**Pull request by byzyk**
As discussed in #2578 we agreed on consistent emphasis style for **bold** and *italic* text.

Such a rule is not a part of [`markdownlint` rules](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md), thus I had to create a [custom rule](https://www.npmjs.com/package/markdownlint-rule-emphasis-style) for our case. Maybe later it can be merged to core package.

### ~Note~

~This PR here is only to add the rule to our config. I'm going to work on another PR later today to fix all remaining issues across the docs.~

### CODE REVIEW

**Comment by byzyk** [comment]
Travis is failing, which is good :)

**Comment by EugeneHlushko** [comment]
Please include replacing all occurrences in this PR instead of new one

**Comment by byzyk** [comment]
Yo @EugeneHlushko, please take a look. I fixed all linter errors and it should be good to go now.

File: src/content/plugins/index.md [hunk 1]
```diff
@@ -7,39 +7,40 @@ contributors:
   - aretecode
   - eko3alpha
   - refactorized
+  - byzyk
 ---
 
-webpack has a rich plugin interface. Most of the features within webpack itself use this plugin interface. This makes webpack **flexible**.
+webpack has a rich plugin interface. Most of the features within webpack itself use this plugin interface. This makes webpack __flexible__.
 
-Name                                                     | Description
```
Comment (EugeneHlushko) [inline src/content/plugins/index.md]: this is not related, please provide motivation for this change or revert

File: src/content/comparison.md [hunk 2]
```diff
@@ -16,36 +17,36 @@ related:
...
+| Feature                                                              | webpack/webpack                                                                | jrburke/requirejs                                | substack/node-browserify                            | jspm/jspm-cli                                                                                                                                     | rollup/rollup                                                                | brunch/brunch                                                                              |
+| -------------------------------------------------------------------- | ------------------------------------------------------------------------------ | ------------------------------------------------ | --------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------ |
```
Comment (EugeneHlushko) [inline src/content/comparison.md]: why table changes? please provide motivation or revert

File: src/content/branding.md [hunk 3]
```diff
@@ -70,19 +71,19 @@ We use the beautiful [Geomanist Medium](http://atipofoundry.com/fonts/geomanist)
...
-| Fiord:        | HEX `#465E69` | `rgb: 70, 94, 105`    | <div style="background-color: #465E69;">&nbsp;</div>
-| Outer Space:  | HEX `#2B3A42` | `rgb: 43, 58, 66`     | <div style="background-color: #2B3A42;">&nbsp;</div>
-| White:        | HEX `#ffffff` | `rgb: 255, 255, 255`  | <div style="background-color: #ffffff;">&nbsp;</div>
-| Concrete:     | HEX `#f2f2f2` | `rgb: 242, 242, 242`  | <div style="background-color: #f2f2f2;">&nbsp;</div>
-| Alto:         | HEX `#dedede` | `rgb: 222, 222, 222`  | <div style="background-color: #dedede;">&nbsp;</div>
-| Dusty Gray:   | HEX `#999999` | `rgb: 153, 153, 153`  | <div style="background-color: #999999;">&nbsp;</div>
-| Dove Gray:    | HEX `#666666` | `rgb: 102, 102, 102`  | <div style="background-color: #666666;">&nbsp;</div>
-| Emperor:      | HEX `#535353` | `rgb: 83, 83, 83`     | <div style="background-color: #535353;">&nbsp;</div>
-| Mine Shaft:   | HEX `#333333` | `rgb: 51, 51, 51`     | <div style="background-color: #333333;">&nbsp;</div>
+| Color Name   | HEX Code      | RGB Code             | Sample                                               |
```
Comment (EugeneHlushko) [inline src/content/branding.md]: same

File: src/content/comparison.md (hunk 2 above)
Comment (byzyk) [reply src/content/comparison.md]: I guess prettier or IDE formated that automatically. It’s more readable in the source code but I will revert later today. Shall we format all tables like this? Maybe in another PR?

**Comment by byzyk** [comment]
I don’t think automatic fix is supported at the moment since the rule is not yet the part of core package, thus prettier doesn’t recognize it

File: src/content/comparison.md (hunk 2 above)
Comment (EugeneHlushko) [reply src/content/comparison.md]: Unfortunately this is not in `.editorconfig` so lets not change formatting of the tables for now.

**Comment by byzyk** [comment]
Hey @EugeneHlushko, reverted the table formatting change and rebased everything on the master. Please take a look 😎

File: src/content/api/cli.md [hunk 4]
```diff
@@ -133,55 +134,55 @@ webpack --env.platform=web  # sets env.platform == "web"
...
-`--output-filename`       | The output filename of the bundle           | string     | [name].js
-`--output-jsonp-function` | The name of the JSONP function used for chunk loading | string | webpackJsonp
-`--output-library`        | Expose the exports of the entry point as library | string |
-`--output-library-target` | The type for exposing the exports of the entry point as library | string | var
-`--output-path`           | The output path for compilation assets      | string     | Current directory
-`--output-pathinfo`       | Include a comment with the request for every dependency | boolean | false
-`--output-public-path`    | The public path for the assets              | string     | /
-`--output-source-map-filename` | The output filename for the SourceMap  | string     | [name].map or [outputFilename].map
-`--build-delimiter` | Display custom text after build output | string | Default string is null. You could provide a string such as `=== Build done ===`
+| Parameter                      | Explanation                                                     | Input type | Default                                                                         |
```
Comment (EugeneHlushko) [inline src/content/api/cli.md]: I still see lots of table changes though, did you accidentally miss these and the rest of table changes?

Comment (byzyk) [reply src/content/api/cli.md]: yep, my bad, diff was collapsed. gonna address this today sorry :)

Comment (byzyk) [reply src/content/api/cli.md]: hey @EugeneHlushko, reverted those missing changes, should be ok now

**Comment by EugeneHlushko** [comment]
Adding @montogeek to reviewers so he can double check

**Comment by byzyk** [comment]
For some reason CLA is pending for me as well, that's weird 🤔

**Comment by EugeneHlushko** [comment]
I think its a problem with github hook status display, i can see it is ok. No worries.
screen shot 2018-11-08 at 10 07 52 am

**Comment by EugeneHlushko** [comment]
Okay i will merge this when tests pass

File: package.json [hunk 5]
```diff
@@ -33,7 +33,7 @@
     "init:generated": "mkdirp ./generated/loaders && mkdirp ./generated/plugins ",
     "lint": "run-s lint:*",
     "lint:js": "eslint . --ext .js,.jsx,.md",
-    "lint:markdown": "markdownlint --config ./.markdownlint.json *.md ./src/content/**/*.md",
+    "lint:markdown": "markdownlint --rules markdownlint-rule-emphasis-style --config ./.markdownlint.json *.md ./src/content/*.md ./src/content/**/*.md",
```
Comment (montogeek) [inline package.json]: It this rules necessary, Shouldn't it get it from the config file?

Comment (EugeneHlushko) [reply package.json]: hmm good question, also i just noticed: `*.md ./src/content/*.md` addition, is this necessary? @byzyk

Comment (byzyk) [reply package.json]: @montogeek please take a look at the PR description, this rule is not the part of `markdownlint` core package yet, I've created a custom rule for this which later can be merged into core if it proves itself valuable. See discussion in DavidAnson/markdownlint#150

@EugeneHlushko I've added `./src/content/*.md` since MD files placed in `content` root (e.g. `content/branding.md`) folder were not covered by existing scope.

Comment (byzyk) [reply package.json]: @montogeek forgot to mention that the actual rule configuration comes from config file. We need `--rule` here just to extend core package with custom rules.

Comment (EugeneHlushko) [reply package.json]: I mean shouldn't `*.md` cover all `.md` files

Comment (byzyk) [reply package.json]: For some reason, files in `content` root folder were not covered. We can attempt to address this in another PR since it's not really related to this one.

The way I discovered this is when `lint-staged` reported some MD linter issues on those files while `npm run lint:markdown` didn't catch those, that's why I explicitly added the root folder for now.

Comment (EugeneHlushko) [reply package.json]: hmm i am fine tackling this in other pr since this already does what it needs to, what do you think @montogeek ?

Comment (montogeek) [reply package.json]: It is fine

**Comment by montogeek** [comment]
@byzyk Just only last thing, did you check this change against `rebuild` branch?

**Comment by byzyk** [comment]
@montogeek hey man, nope I haven't. We can merge this one and then merge it with `rebuild` and see if there are any conflicts but I doubt that since we just extended config slightly.

**Comment by EugeneHlushko** [comment]
yup should be good, also this package @byzyk created should soon afterwards be available as part of main configuration. So by the time we do rebuild merge it could be even simpler

**Comment by EugeneHlushko** [comment]
updating branch just in case to re-run against latest master and if good will merge

**Comment by EugeneHlushko** [comment]
Thanks! Now lets keep an eye for that pattern match `*.md`
